packages/swayze/lib/src/widgets/headers/header_drag_and_drop_preview.dart

```diff
@@ -0,0 +1,325 @@
+        )
+        .pixel;
+
+    final blockedRange = Range(
+      dragState.headers.start,
+      dragState.headers.end + 1,
+    );
+
+    return Stack(
+      children: [
+        _PreviewRect(
+          axis: axis,
+          pointerPosition: dragState.position,
+          headerPosition: headerPosition,
+          headerExtent: headerExtent,
+          color: swayzeStyle.dragNDropPreviewHeadersColor,
+        ),
+        if (!blockedRange.contains(currentHeaderIndex))
+          _PreviewLine(
+            axis: axis,
+            lineColor: lineColor,
+            lineWidth: lineWidth,
+            dropHeaderAtPosition: dropHeaderAtPosition,
+          ),
+      ],
+    );
+  }
+}
+
+class _PreviewLine extends LeafRenderObjectWidget {
+  final Color lineColor;
+
+  final double lineWidth;
+
+  final double dropHeaderAtPosition;
+
+  final Axis axis;
+
+  const _PreviewLine({
+    Key? key,
+    required this.lineColor,
+    required this.lineWidth,
+    required this.dropHeaderAtPosition,
+    required this.axis,
+  }) : super(key: key);
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderPreviewLine(
+      axis,
+      lineColor,
+      lineWidth,
+      dropHeaderAtPosition,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+    BuildContext context,
+    _RenderPreviewLine renderObject,
+  ) {
+    renderObject
+      ..axis = axis
+      ..lineWidth = lineWidth
+      ..lineColor = lineColor
+      ..dropHeaderAtPosition = dropHeaderAtPosition;
+  }
+}
+
+class _RenderPreviewLine extends RenderBox {
+  Color _lineColor;
+
+  Color get lineColor => _lineColor;
+
+  set lineColor(Color value) {
+    _lineColor = value;
+    markNeedsPaint();
+  }
+
+  double _lineWidth;
+
+  double get lineWidth => _lineWidth;
+
+  set lineWidth(double value) {
+    _lineWidth = value;
+    markNeedsLayout();
+  }
+
+  double _dropHeaderAtPosition;
+  double get dropHeaderAtPosition => _dropHeaderAtPosition;
+  set dropHeaderAtPosition(double value) {
+    _dropHeaderAtPosition = value;
+    markNeedsLayout();
+  }
+
+  Axis _axis;
+  Axis get axis => _axis;
+  set axis(Axis value) {
+    _axis = value;
+    markNeedsLayout();
+  }
+
+  _RenderPreviewLine(
+    this._axis,
+    this._lineColor,
+    this._lineWidth,
+    this._dropHeaderAtPosition,
+  );
+
+  @override
+  bool get alwaysNeedsCompositing => true;
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    return constraints.biggest;
+  }
+
+  late final linePaintCache = CachedValue(
+    () {
+      return Paint()
+        ..color = lineColor
+        ..strokeWidth = lineWidth;
+    },
+  ).withDependency<Color?>(() => lineColor).withDependency(() => lineWidth);
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final canvas = context.canvas;
+    canvas.translate(-0.5, -0.5);
```

Comment:
shouldnt this come after the canvas.save?

Reply:
nop. This is the same logic from other table lines, and they all come before canvas.save, it may be related to this issue 🤔